src/main/java/org/jenkinsci/plugins/codesonar/CodeSonarPublisher.java

```diff
@@ -479,188 +569,107 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @Nonnul
-    }
-
-    public void setHttpService(HttpService httpService) {
-        this.httpService = httpService;
-    }
-    
-    public void setAuthenticationService(AuthenticationService authenticationService) {
-        this.authenticationService = authenticationService;
-    }
-
-    public String getCredentialId() {
-        return credentialId;
-    }
-
-    public void setCredentialId(String credentialId) {
-        this.credentialId = credentialId;
-    }
-
-    public HttpService getHttpService(@Nonnull Run<?, ?> run) throws CodeSonarPluginException {
-        if (httpService == null) {
-            Collection<? extends Certificate> serverCertificates = null;
-            if(StringUtils.isNotEmpty(serverCertificateCredentialId)) {
-                serverCertificateCredentials = CredentialsMatchers.firstOrNull(
-                        CredentialsProvider.lookupCredentials(StandardCredentials.class, run.getParent(), ACL.SYSTEM,
-                                Collections.<DomainRequirement>emptyList()), CredentialsMatchers.withId(getServerCertificateCredentialId()));
-    
-                if(serverCertificateCredentials instanceof FileCredentials) {
-                    LOGGER.log(Level.INFO, "Found FileCredentials provided as Hub HTTPS certificate");
-                    FileCredentials f = (FileCredentials) serverCertificateCredentials;
-                    try {
-                        CertificateFactory cf = CertificateFactory.getInstance("X.509");
-                        serverCertificates = cf.generateCertificates(f.getContent());
-                        LOGGER.log(Level.INFO, "X509Certificate initialized");
-                    } catch (IOException | CertificateException e ) {
-                        throw createError("Failed to create X509Certificate from Secret File Credential.", e);
-                    }
-                } else {
-                    if(serverCertificateCredentials != null) {
-                        LOGGER.log(Level.INFO, "Found {0} provided as Hub HTTPS certificate", serverCertificateCredentials.getClass().getName());
-                        throw createError("The Jenkins Credentials provided as Hub HTTPS certificate is of type {0}.%nPlease provide a credential of type FileCredentials", serverCertificateCredentials.getClass().getName());
-                    }
-                    LOGGER.log(Level.INFO, "Credentials with id \"{0}\" not found", getServerCertificateCredentialId());
-                    throw createError(CodeSonarLogger.formatMessage("Credentials with id \"{0}\" not found", getServerCertificateCredentialId()));
+    private HttpService createHttpService(@Nonnull Run<?, ?> run) throws CodeSonarPluginException {
+        Collection<? extends Certificate> serverCertificates = null;
+        if (StringUtils.isNotEmpty(serverCertificateCredentialId)) {
+            serverCertificateCredentials = lookupCredentials(
+                getServerCertificateCredentialId(),
+                run);
```

Comment:
These lines should either use `this.serverCertificateCredentialId` or `this.getServerCertificateCredentialId()`, but not both.

Reply:
Done.  Removed serverCertificateCredentials field.